net/query_filter/query_filter.cc

```diff
@@ -133,3 +137,38 @@ absl::optional<GURL> ApplyQueryFilter(const GURL& original_url) {
   }
   return absl::nullopt;
 }
+
+bool ShouldFilterTwitterParam(
+    const GURL& url_to_load,
+    const absl::optional<url::Origin>& request_initiator) {
+  return url_to_load.DomainIs("twitter.com") &&
+         (!request_initiator ||
+          !request_initiator.value().DomainIs("twitter.com")) &&
+         url_to_load.has_query();
+}
+
+GURL RemoveTwitterParam(const GURL& url_to_load) {
```

Comment:
I wonder if you can generalize `StripQueryParameter` to handle twitter and other cases?

Reply:
Yes, but I'm keeping it separate intentionally in this PR since it's kind of an experimental fix for Twitter for now. My intention is to come back to this and merge the two together after this has had more testing.